code/datums/spell.dm

```diff
@@ -3,9 +3,53 @@
 
 /obj/effect/proc_holder
 	var/panel = "Debug"//What panel the proc holder needs to go on.
+	var/active = FALSE //Used by toggle based abilities.
+	var/ranged_mousepointer
 
 var/list/spells = typesof(/obj/effect/proc_holder/spell) //needed for the badmin verb for now
 
+/obj/effect/proc_holder/proc/InterceptClickOn(mob/living/user, params, atom/A)
+	if(user.ranged_ability != src)
+		to_chat(user, "<span class='warning'><b>[user.ranged_ability.name]</b> has been disabled.")
+		user.ranged_ability.remove_ranged_ability(user)
+		return TRUE //TRUE for failed, FALSE for passed.
+	user.next_click = world.time + CLICK_CD_CLICK_ABILITY

```

Comment:
The next_click var is checked right after the ranged click intercept (and the spells use the intercept), whereas the next_move var is checked after the key + click handling (which shouldn't be fired). I imagine that's the reason /tg/ opted to use next_click.